src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,134 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private int size;
+    private final float loadFactor;
+    private Node<K, V>[] table;
+    private int capacity;
+
+    public MyHashMap() {
+        this.capacity = 16;
+        this.loadFactor = 0.75f;
+        this.size = 0;
+    }
 
     @Override
-    public void put(K key, V value) {
+    public V put(K key, V value) {
+
+        if (table == null || size >= capacity * loadFactor) {
+            resize();
+        }
+
+        if (key == null) {
+            if (table[0] == null) {
+                table[0] = new Node<>(null, value, null);
+                size++;
+            } else {
+                Node<K, V> currentNode = table[0];
+                while (currentNode != null) {
+                    if (currentNode.key == null) {
+                        currentNode.value = value;
+                        break;
+                    }
+                    currentNode = currentNode.next;
+                }
+                if (currentNode == null) {
+                    table[0].addNode(null, value);
+                    size++;
+                }
+            }
+            return value;
+        }
+
+        int hash = hash(key);
+        int index;
+        index = hash & (table.length - 1);
+
+        if (table[index] == null) {
+            table[index] = new Node<>(key, value, null);
+            size++;
+        } else {
+            Node<K, V> currentNode;
+            for (currentNode = table[index]; currentNode.next != null;
```

Comment:
too big method, simplify it, you don't need a separate block of code for null key